### PR TITLE
Update compatibility data output

### PR DESCRIPTION
**Pull request by jessepearson**
Fixes https://github.com/Automattic/woocommerce-payments-server/issues/5532

#### Changes proposed in this Pull Request

* Update the `get_option` call for `woocommerce_permalinks` to have a default of an empty array so that an array is always sent.
  * This should never be an issue due to WooCommerce core itself updates the permalinks for WC, even if the admin never does.
* Update the calls for the `shop`, `cart`, and `checkout` permalinks to check for possible `false` values returned, and instead return `Not set`. 

#### Testing instructions

* Check out this branch.
* Navigate to WooCommerce > Settings > Products.
* Click the X to clear your _Shop page_ settings, and then save.
* Navigate to WooCommerce > Settings > Advanced.
* Click the X to clear your _Cart page_ and _Checkout page_ settings, and then save.
* Navigate to WCPay Dev Tools and clear your account cache.
* Check your `logstash` for your local server and look for the latest `compatibility` API request (it should be the last thing in the log`.
* The values for the permalinks should look like the example below.
* Don't forget to reset your permalinks you cleared above.

```
"woocommerce_shop": "Not set",
"woocommerce_cart": "Not set",
"woocommerce_checkout": "Not set",
```

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8935` or branch name `fix/5532-fix-compatibility-data` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/5532-fix-compatibility-data) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 4e158e4e8d895cfca5c93c99cc86ded47b11c8b3
- Build time: 2024-06-18 14:24:06 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/class-compatibility-service.php [hunk 1]
```diff
@@ -76,10 +76,10 @@ public function add_compatibility_onboarding_data( $args ): array {
 	private function get_compatibility_data(): array {
 		$active_plugins        = get_option( 'active_plugins', [] );
 		$post_types_count      = $this->get_post_types_count();
-		$wc_permalinks         = get_option( 'woocommerce_permalinks' );
-		$wc_shop_permalink     = get_permalink( wc_get_page_id( 'shop' ) );
-		$wc_cart_permalink     = get_permalink( wc_get_page_id( 'cart' ) );
-		$wc_checkout_permalink = get_permalink( wc_get_page_id( 'checkout' ) );
+		$wc_permalinks         = get_option( 'woocommerce_permalinks', [] );
+		$wc_shop_permalink     = get_permalink( wc_get_page_id( 'shop' ) ) ? get_permalink( wc_get_page_id( 'shop' ) ) : 'Not set';
```
Comment (eduardoumpierre) [inline includes/class-compatibility-service.php]: What do you think about extracting that repeating logic to a method? My idea is to have something like this:

<details><summary>Details</summary>

```diff
diff --git a/includes/class-compatibility-service.php b/includes/class-compatibility-service.php
index bd831fd52..6a66941a1 100644
--- a/includes/class-compatibility-service.php
+++ b/includes/class-compatibility-service.php
@@ -77,9 +77,9 @@ class Compatibility_Service {
 		$active_plugins        = get_option( 'active_plugins', [] );
 		$post_types_count      = $this->get_post_types_count();
 		$wc_permalinks         = get_option( 'woocommerce_permalinks', [] );
-		$wc_shop_permalink     = get_permalink( wc_get_page_id( 'shop' ) ) ? get_permalink( wc_get_page_id( 'shop' ) ) : 'Not set';
-		$wc_cart_permalink     = get_permalink( wc_get_page_id( 'cart' ) ) ? get_permalink( wc_get_page_id( 'cart' ) ) : 'Not set';
-		$wc_checkout_permalink = get_permalink( wc_get_page_id( 'checkout' ) ) ? get_permalink( wc_get_page_id( 'checkout' ) ) : 'Not set';
+		$wc_shop_permalink     = $this->get_permalink_for_page_id( 'shop' );
+		$wc_cart_permalink     = $this->get_permalink_for_page_id( 'cart' );
+		$wc_checkout_permalink = $this->get_permalink_for_page_id( 'checkout' );
 
 		return [
 			'woopayments_version'    => WCPAY_VERSION_NUMBER,
@@ -114,4 +114,17 @@ class Compatibility_Service {
 
 		return $post_types_count;
 	}
+
+	/**
+	 * Gets the permalink for a page ID.
+	 *
+	 * @param string $page_id The page ID to get the permalink for.
+	 *
+	 * @return string The permalink for the page ID, or 'Not set' if the permalink is not available.
+	 */
+	private function get_permalink_for_page_id( string $page_id ): string {
+		$permalink = get_permalink( wc_get_page_id( $page_id ) );
+
+		return $permalink ? $permalink : 'Not set';
+	}
 }

```
</details>

**Review by eduardoumpierre — APPROVED** [review]
The changes make sense and the code is covered by unit tests. I've followed the testing instructions and everything worked as expected.

I left a comment that I believe would improve the readability of the `get_compatibility_data` method, but since it's not a hard blocker, I'm pre-approving the PR.

Thanks for working on this issue! LGTM :shipit:

File: includes/class-compatibility-service.php (hunk 1 above)
Comment (jessepearson) [reply includes/class-compatibility-service.php]: Just applied this in https://github.com/Automattic/woocommerce-payments/pull/8935/commits/4e158e4e8d895cfca5c93c99cc86ded47b11c8b3. It's a great suggestion, thank you.

**Pull request merged**